### PR TITLE
Big PickupAPI changes (Breaking)

### PR DESCRIPTION
**Pull request by IRacle1**
- Better property saving between items/pickups.projectiles (without IsLoaded)
 - Armor `MovementSpeedMultiplier` setter was removed due desync
 - Optimization in `CreatePickup` method + now it creates item with exactly same serial
 - Pickup::Spawn is non static now, i think this is more logical and understandable
 - `Player::AddItem` now has more overloads with `FirearmType` and attachments

### CODE REVIEW

File: Exiled.API/Features/Pickups/Pickup.cs [hunk 1]
```diff
@@ -492,26 +491,7 @@ public static IEnumerable<T> Get<T>(IEnumerable<GameObject> gameObjects)
...
-        /// <seealso cref="Projectile.Spawn(Projectile, Vector3, Quaternion, bool, Player)"/>
-        public static Pickup Spawn(Pickup pickup, Vector3 position, Quaternion rotation, Player previousOwner = null)
-        {
-            pickup.Position = position;
-            pickup.Rotation = rotation;
-            pickup.PreviousOwner = previousOwner;
-            pickup.Spawn();
-
-            return pickup;
-        }
```
Comment (louis1706) [inline Exiled.API/Features/Pickups/Pickup.cs:514]: this method can keep at Obsolete

File: Exiled.API/Features/Pickups/Projectiles/Projectile.cs [hunk 2]
```diff
@@ -101,35 +101,34 @@ internal Projectile(ItemType type)
...
         /// <summary>
         /// Spawns a <see cref="Projectile"/>.
         /// </summary>
-        /// <param name="pickup">The <see cref="Projectile"/> too spawn.</param>
         /// <param name="position">The position to spawn the <see cref="Projectile"/> at.</param>
         /// <param name="rotation">The rotation to spawn the <see cref="Projectile"/>.</param>
         /// <param name="shouldBeActive">Whether the <see cref="Projectile"/> should be in active state after spawn.</param>
         /// <param name="previousOwner">An optional previous owner of the item.</param>
-        /// <returns>The <see cref="Projectile"/> Spawn.</returns>
-        public static Projectile Spawn(Projectile pickup, Vector3 position, Quaternion rotation, bool shouldBeActive = true, Player previousOwner = null)
```
Comment (louis1706) [inline Exiled.API/Features/Pickups/Projectiles/Projectile.cs:115]: a copy of the Static method can be keep at Obsolete and call the new one

File: Exiled.API/Features/Player.cs [hunk 3]
```diff
@@ -2198,17 +2208,27 @@ public void SetAmmo(AmmoType ammoType, ushort amount)
...
+            return Item.Get(Inventory.ServerAddItem(itemType));
+        }
+
+        /// <summary>
+        /// Adds an firearm of the specified type with default durability(ammo/charge) and no mods to the player's inventory.
+        /// </summary>
+        /// <param name="firearmType">The firearm to be added.</param>
         /// <param name="identifiers">The attachments to be added to the item.</param>
         /// <returns>The <see cref="Item"/> given to the player.</returns>
-        public Item AddItem(ItemType itemType, IEnumerable<AttachmentIdentifier> identifiers = null)
```
Comment (louis1706) [inline Exiled.API/Features/Player.cs:2203]: we can make a copy of this method at Obsolete to call the new one

File: Exiled.API/Features/Player.cs [hunk 4]
```diff
@@ -2241,13 +2261,13 @@ public IEnumerable<Item> AddItem(ItemType itemType, int amount)
...
         /// <summary>
-        /// Adds the amount of items of the specified type with default durability(ammo/charge) and no mods to the player's inventory.
+        /// Adds the amount of firearms of the specified type with default durability(ammo/charge) and no mods to the player's inventory.
         /// </summary>
-        /// <param name="itemType">The item to be added.</param>
+        /// <param name="firearmType">The item to be added.</param>
         /// <param name="amount">The amount of items to be added.</param>
         /// <param name="identifiers">The attachments to be added to the item.</param>
         /// <returns>An <see cref="IEnumerable{Item}"/> containing the items given.</returns>
-        public IEnumerable<Item> AddItem(ItemType itemType, int amount, IEnumerable<AttachmentIdentifier> identifiers)
```
Comment (louis1706) [inline Exiled.API/Features/Player.cs:2250]: we can make a copy of this method at Obsolete to call the new one

File: Exiled.API/Features/Player.cs [hunk 5]
```diff
@@ -2284,11 +2304,11 @@ public IEnumerable<Item> AddItem(IEnumerable<ItemType> items)
         /// </summary>
         /// <param name="items">The <see cref="Dictionary{TKey, TValue}"/> of <see cref="ItemType"/> and <see cref="IEnumerable{T}"/> of <see cref="AttachmentIdentifier"/> to be added.</param>
         /// <returns>An <see cref="IEnumerable{Item}"/> containing the items given.</returns>
-        public IEnumerable<Item> AddItem(Dictionary<ItemType, IEnumerable<AttachmentIdentifier>> items)
```
Comment (louis1706) [inline Exiled.API/Features/Player.cs:2287]: we can make a copy of this method at Obsolete to call the new one

File: Exiled.API/Features/Player.cs [hunk 6]
```diff
@@ -2401,35 +2416,6 @@ public Item AddItem(ItemBase itemBase, Item item = null)
...
-        /// <param name="identifiers">The attachments to be added to the item.</param>
-        public void AddItem(Firearm firearm, int amount, IEnumerable<AttachmentIdentifier> identifiers)
-        {
-            if (amount > 0)
-            {
-                for (int i = 0; i < amount; i++)
-                    AddItem(firearm, identifiers);
-            }
-        }
-
```
Comment (louis1706) [inline Exiled.API/Features/Player.cs:2432]: Make this Obsolete

File: Exiled.Events/EventArgs/Map/ChangedIntoGrenadeEventArgs.cs [hunk 7]
```diff
@@ -43,11 +39,5 @@ public ChangedIntoGrenadeEventArgs(TimedGrenadePickup pickup, ThrownProjectile p
         /// Gets a value indicating the projectile that spawned.
         /// </summary>
         public Projectile Projectile { get; }
-
-        /// <summary>
-        /// Gets or sets a value indicating how long the fuse of the changed grenade will be.
-        /// </summary>
-        // TODO: float
-        public double FuseTime { get; set; }
```
Comment (louis1706) [inline Exiled.Events/EventArgs/Map/ChangedIntoGrenadeEventArgs.cs:51]: We need to make this Obsolete

**Review by louis1706 — CHANGES_REQUESTED** [review]

**Comment by louis1706** [comment]
this actually not a lot of Breaking change

**Review by louis1706 — APPROVED** [review]

**Review by louis1706 — APPROVED** [review]

**Review by Misaka-ZeroTwo — APPROVED** [review]

**Review by Misaka-ZeroTwo — APPROVED** [review]

**Review by louis1706 — APPROVED** [review]

**Pull request merged**